crates/apollo-compiler/src/database/hir_db.rs

```diff
@@ -34,75 +35,172 @@ pub trait HirDatabase: InputDatabase + AstDatabase {
 
+    /// Return any type definitions that contain the corresponding directive
     fn find_types_with_directive(&self, directive: String) -> Arc<Vec<TypeDefinition>>;
 
+    /// Return an input object type definition corresponding to the name.
+    /// Result of this query is not cached internally.
+    #[salsa::transparent]
     fn find_input_object_by_name(&self, name: String) -> Option<Arc<InputObjectTypeDefinition>>;
 
     fn types_definitions_by_name(&self) -> Arc<IndexMap<String, TypeDefinition>>;
 
+    /// Return a type definition corresponding to the name.
+    /// Result of this query is not cached internally.
+    #[salsa::transparent]
     fn find_type_definition_by_name(&self, name: String) -> Option<TypeDefinition>;
 
+    /// Return all query operations in a corresponding file.
     fn query_operations(&self, file_id: FileId) -> Arc<Vec<Arc<OperationDefinition>>>;
 
+    /// Return all mutation operations in a corresponding file.
     fn mutation_operations(&self, file_id: FileId) -> Arc<Vec<Arc<OperationDefinition>>>;
 
+    /// Return all subscription operations in a corresponding file.
     fn subscription_operations(&self, file_id: FileId) -> Arc<Vec<Arc<OperationDefinition>>>;
 
+    /// Return all operation fields in a corresponding selection set.
     fn operation_fields(&self, selection_set: SelectionSet) -> Arc<Vec<Field>>;
 
+    /// Return all operation inline fragment fields in a corresponding selection set.
     fn operation_inline_fragment_fields(&self, selection_set: SelectionSet) -> Arc<Vec<Field>>;
 
+    /// Return all operation fragment spread fields in a corresponding selection set.
     fn operation_fragment_spread_fields(&self, selection_set: SelectionSet) -> Arc<Vec<Field>>;
 
+    /// Return all variables in a corresponding selection set.
     fn selection_variables(&self, selection_set: SelectionSet) -> Arc<HashSet<Variable>>;
 
+    /// Return all variables in corresponding variable definitions.
     fn operation_definition_variables(
         &self,
         variables: Arc<Vec<VariableDefinition>>,
     ) -> Arc<HashSet<Variable>>;
 
+    /// Return a subtype map of the current compiler's type system.
+    ///
+    /// Given the following schema,
+    /// ```graphql
+    /// type Query {
+    ///   me: String
+    /// }
+    /// type Foo {
+    ///   me: String
+    /// }
+    /// type Bar {
+    ///   me: String
+    /// }
+    /// union UnionType = Foo | Bar
+    ///
+    /// interface Baz {
+    ///   me: String,
+    /// }
+    /// type ObjectType implements Baz { me: String }
+    /// interface InterfaceType implements Baz { me: String }
+    /// ```
+    /// we can say that:
+    ///
+    /// - `Foo` and `Bar` are a subtypes of `UnionType`.
+    /// - `ObjectType` and `InterfaceType` are subtypes of `Baz`.
     fn subtype_map(&self) -> Arc<HashMap<String, HashSet<String>>>;
 
+    /// Return `true` if the provided `abstract_type` is a subtype of the
+    /// corresponding `maybe_subtype`.
```

Comment:
This sentence seems backwards?

```suggestion
    /// Return `true` if the provided `maybe_subtype` is a subtype of the
    /// corresponding `abstract_type`.
```